### PR TITLE
Removing/Replacing all unnecessary CMake variables

### PR DESCRIPTION
**Pull request by jameskr97**
This is my fourth pull request on this repository.

Here is a summary of everything changed in this pull request.

## Main Goals
- [x] Remove `WITH_NETWORKING` and related variables
- [x] Remove `WITH_JPEG` and related variables
- [x] Remove `WITH_PORTABLE_TOMCRYPT`
- [x] Remove `WITH_GPL_LIBS`
- [x] Remove `WITH_WAV` and `HAS_WAV`
- [x] Remove `WITH_MP3` and `HAS_MP3`
- [x] Remove `WITH_OGG` and `HAS_OGG`
- [x] Remove `WITH_TTY`
- [x] Remove `BSD`
- [x] Remove `CMAKE_POWERED`
- [x] Replace `WIN32`, `WINDOWS`, `_WINDOWS`, `UNIX`, `LINUX`, and `MACOSX` with proper, non-cmake defined pre-processor directives.
- [x] Remove unreferenced variables like `WITH_PROFILING`, `WITH_MINIMAID`, `WITH_PARALLEL_PORT`, and `WITH_IRC_POST_HOOK`

## Requested 
- [x] Remove `ezsockets.h/cpp`
- [x] ~~Remove unnecessary`ver.h` in place of `verstub.in.h`~~ Note: This will wait for the CMake rewrite.

### CODE REVIEW

**Comment by jameskr97** [comment]
Note: Removing the options for these modules makes them required, though that's the intention due to the nature of the game. They were always required anyway.

**Comment by jameskr97** [comment]
Ready for review in it's current state. There are some other variables which are unnecessary and will be removed, but require more CMake refactoring.

File: src/Etterna/Globals/StdString.h [hunk 1]
```diff
@@ -51,7 +51,7 @@
 //			- Jim Cline
 //			- Jeff Kohn
 //			- Todd Heckel
-//			- Ullrich Poll�hne
+//			- Ullrich Poll�hne
```
Comment (martensm) [inline src/Etterna/Globals/StdString.h]: Undo mojibake here.
```suggestion
//			- Ullrich Pollähne
```

File: src/Etterna/Globals/StdString.h [hunk 2]
```diff
@@ -500,7 +500,7 @@ class CStdStr : public std::basic_string<CT>
 		// <nChars> or the NULL terminator, whichever comes first.  Since we
 		// are about to call a less forgiving overload (in which <nChars>
 		// must be a valid length), we must adjust the length here to a safe
-		// value.  Thanks to Ullrich Poll�hne for catching this bug
+		// value.  Thanks to Ullrich Poll�hne for catching this bug
```
Comment (martensm) [inline src/Etterna/Globals/StdString.h]: Undo mojibake here.
```suggestion
		// value.  Thanks to Ullrich Pollähne for catching this bug
```

File: src/Etterna/Globals/StdString.h [hunk 3]
```diff
@@ -528,7 +528,7 @@ class CStdStr : public std::basic_string<CT>
 		// <nChars> or the NULL terminator, whichever comes first.  Since we
 		// are about to call a less forgiving overload (in which <nChars>
 		// must be a valid length), we must adjust the length here to a safe
-		// value. Thanks to Ullrich Poll�hne for catching this bug
+		// value. Thanks to Ullrich Poll�hne for catching this bug
```
Comment (martensm) [inline src/Etterna/Globals/StdString.h]: Undo mojibake here.
```suggestion
		// value. Thanks to Ullrich Pollähne for catching this bug
```

**Comment by jameskr97** [comment]
Please wait until I can fix the Appveyor CI showing. I only tested this on linux and prematurely marked this PR ready for review. Will update soon, as I am currently working on this.

**Comment by jameskr97** [comment]
Reverting commits outside the scope of this PR. Windows build issue to be resolved in overall cmake-remake.

**Comment by jameskr97** [comment]
This is ready to be merged in its current state.

**Comment by nico-abram** [comment]
LGTM

**Pull request merged**